### PR TITLE
Fix sidecar injector default template config option.

### PR DESCRIPTION
**Pull request by bianpengyuan**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: manifests/charts/istio-control/istio-discovery/templates/istiod-injector-configmap.yaml [hunk 1]
```diff
@@ -21,9 +21,11 @@ data:
   # to fine tune it or use it with kube-inject.
   config: |-
     # defaultTemplates defines the default template to use for pods that do not explicitly specify a template
-    # NOTE: this currently only allows a single element
     {{- if .Values.sidecarInjectorWebhook.defaultTemplates }}
-    defaultTemplates: {{ .Values.sidecarInjectorWebhook.defaultTemplates }}
+    defaultTemplates:
+{{- range .Values.sidecarInjectorWebhook.defaultTemplates}}
```
Comment (howardjohn) [inline manifests/charts/istio-control/istio-discovery/templates/istiod-injector-configmap.yaml:26]: I thought a list turned into a list?

```
$ ik manifest generate -d manifests --set .values.sidecarInjectorWebhook.defaultTemplates[0]=bar --force | grep defaultTemplates
        "defaultTemplates": [
    # defaultTemplates defines the default template to use for pods that do not explicitly specify a template
    defaultTemplates: [bar]
```

Seems to work ok, although I had to put `--force`

**Review by howardjohn — APPROVED** [review]

File: manifests/charts/istio-control/istio-discovery/templates/istiod-injector-configmap.yaml (hunk 1 above)
Comment (bianpengyuan) [reply manifests/charts/istio-control/istio-discovery/templates/istiod-injector-configmap.yaml:26]: If you specify two values, it becomes:
```
istioctl manifest generate --set .values.sidecarInjectorWebhook.defaultTemplates[0]=bar --set .values.sidecarInjectorWebhook.defaultTemplates[1]=foo --force | grep defaultTemplates
        "defaultTemplates": [
    # defaultTemplates defines the default template to use for pods that do not explicitly specify a template
    defaultTemplates: [bar foo]
```
and istiod unmarshals that to a single string (`bar foo`)..

**Comment by howardjohn** [comment]
huh, I think that is a bug. I wonder if it would be fixed by defining
defaultTemplates in value_types.proto so it knows its a list?

**Comment by bianpengyuan** [comment]
> huh, I think that is a bug. I wonder if it would be fixed by defining
defaultTemplates in value_types.proto so it knows its a list?

So defining it as a repeated string is not enough? https://github.com/istio/istio/blob/cfcbcd9a04f34f2397a6cbf52db77901962bf5eb/operator/pkg/apis/istio/v1alpha1/values_types.proto#L999

**Comment by howardjohn** [comment]
nevermind, ignore me, I had an old version. LGTM

**Review by ostromart — APPROVED** [review]

**Pull request merged**